File: kotlin/leetcode_koltin/src/FizzBuzz412.kt
class FizzBuzz412(private val limit: Int) {
    fun fizzBuzzWithForLoop() {
        val output = mutableListOf<String>()
        println("Using for loop:")
        for (n in 1..limit) {
            when {
                n % 3 == 0 && n % 5 == 0 -> output.add("FizzBuzz")
                n % 3 == 0 -> output.add("Fizz")
                n % 5 == 0 -> output.add("Buzz")
                else -> output.add(n.toString())
            }
        }
        println(output)
    }

    fun fizzBuzzWithARange() {
        val output = mutableListOf<String>()
        println("Using a range:")
        // output.clear() // Clear the previous output
        for (n in 1..limit) {
            output.add(
                when {
                    n % 3 == 0 && n % 5 == 0 -> "FizzBuzz"
                    n % 3 == 0 -> "Fizz"
                    n % 5 == 0 -> "Buzz"
                    else -> n.toString()
                }
            )
        }
        println(output + "\n")
    }
}

File: ruby/leetcode_ruby/fizz_buzz412.rb
class FizzBuzz412
    def initialize(limit)
        @limit = limit
    end

    def calculate_fizz_buzz()
        output = []

        # create a list then join it
        for n in 1..@limit
            if n % 3 == 0 and n % 5 == 0
                output.append('FizzBuzz')
            elsif n % 5 == 0
                output.append('Buzz')
            elsif n % 3 == 0
                output.append('Fizz')
            else
                output.append(n.to_s)
            end
        end

        puts "join: " + output.join(', ') + "\n"

        # map creates a list
        output = (1..@limit).map do |n|
          if n % 3 == 0 && n % 5 == 0
            "FizzBuzz"
          elsif n % 3 == 0
            "Fizz"
          elsif n % 5 == 0
            "Buzz"
          else
            n.to_s
          end
        end

        return output
    end
end
